code/modules/vehicles/motorized_wheelchair.dm

```diff
@@ -83,35 +90,35 @@
 			to_chat(user, "<span class='notice'>You install the [I].</span>")
 		refresh_parts()
 		return
-	if(istype(I, /obj/item/stock_parts))
-		var/obj/item/stock_parts/B = I
-		var/P
-		for(var/obj/item/stock_parts/A in contents)
-			for(var/D in required_parts)
-				if(ispath(A.type, D))
-					P = D
-					break
-			if(istype(B, P) && istype(A, P))
-				if(B.get_part_rating() > A.get_part_rating())
-					B.forceMove(src)
-					user.put_in_hands(A)
-					user.visible_message("<span class='notice'>[user] replaces [A] with [B] in [src].</span>", "<span class='notice'>You replace [A] with [B].</span>")
-					break
-		refresh_parts()
-		return
-	return ..()
+	if(!istype(I, /obj/item/stock_parts))
+		return ..()
+
+	var/obj/item/stock_parts/newstockpart = I
+	var/type_to_check
+	for(var/obj/item/stock_parts/oldstockpart in contents)
+		for(var/pathtypes in required_parts)
+			if(ispath(oldstockpart.type, pathtypes))
```

Comment:
Why not put type_to_check a layer down? it's not used outside this loop

Reply:
Done